process/transactionLog/process.go

```diff
@@ -59,36 +57,54 @@ func (tlp *txLogProcessor) GetLog(txHash []byte) (data.LogHandler, error) {
 		return nil, process.ErrLogNotFound
 	}
 
-	var txLog data.LogHandler
-	err = tlp.marshalizer.Unmarshal(txLog, txLogBuff)
+	var txLog transaction.Log
+	err = tlp.marshalizer.Unmarshal(&txLog, txLogBuff)
 	if err != nil {
 		return nil, err
 	}
 
-	return txLog, nil
+	return &txLog, nil
 }
 
 // GetLogFromCache retrieves a log generated by a transaction from ram
 func (tlp *txLogProcessor) GetLogFromCache(txHash []byte) (data.LogHandler, bool) {
```

Comment:
the comment on the function is different from what the actual implementation does. I think only GetLog function should be enough, which tries cache first than reads from storage.

Reply:
changed the comment and added a TODO to remove this function ( in order to remove this function we need also to remove it from the elastic-indexer-go )
It will be fixed in the next PR